File: 1.two-sum.cs
using System; 
using System.Collections.Generic;   
/*
 * @lc app=leetcode id=1 lang=csharp
 *
 * [1] Two Sum
 */

// @lc code=start
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var dic = new Dictionary<int, int>();
        for ( int i=0; i<nums.Length; i++) {
            int key = target - nums[i];
            if (!dic.ContainsKey(key)) {
                dic.Add(key, i);
            }
        }
        for(int i=0; i<nums.Length; i++) {
            if(dic.ContainsKey(nums[i])){
                return new int[] { i, dic[nums[i]] };
            }
        }
        return new int[];
    }
}
// @lc code=end

File: 1.two-sum.js
var twosum = function(num2, target) {
  const obj = {};
  for (var i=0; i< num2.length; i++) {
    if(obj[num2[i]] !== undefined) return [obj[num2[i]], i];
    obj[target-num2[i]] = i;
  }

}

var num2 = [2,7,5,15];
var target = 9;
console.log(twosum(num2, target));